code/parachain/runtime/dali/src/lib.rs

```diff
@@ -1196,6 +1211,157 @@ impl cosmwasm::Config for Runtime {
 	type WeightInfo = cosmwasm::weights::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+	pub const RelayChainId: RelayChain = RelayChain::Rococo;
+	pub const SpamProtectionDeposit: Balance = 1_000_000_000_000;
+	pub const MinimumConnectionDelay: u64 = 0; // well we don't need the security tbh.
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub struct Router {
+	pallet_ibc_ping: pallet_ibc_ping::IbcModule<Runtime>,
+}
+
+impl ModuleRouter for Router {
+	fn get_route_mut(&mut self, module_id: &impl Borrow<ModuleId>) -> Option<&mut dyn Module> {
+		match module_id.borrow().to_string().as_str() {
+			pallet_ibc_ping::MODULE_ID => Some(&mut self.pallet_ibc_ping),
+			_ => None,
+		}
+	}
+
+	fn has_route(module_id: &impl Borrow<ModuleId>) -> bool {
+		matches!(module_id.borrow().to_string().as_str(), pallet_ibc_ping::MODULE_ID)
+	}
+
+	fn lookup_module_by_port(port_id: &PortId) -> Option<ModuleId> {
+		match port_id.as_str() {
+			pallet_ibc_ping::PORT_ID => ModuleId::from_str(pallet_ibc_ping::MODULE_ID).ok(),
+			_ => None,
+		}
+	}
+}
+
+pub struct IbcDenomToAssetIdConversion;
+
+impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
+	type Error = DispatchError;
+
+	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
+		let denom_bytes = denom.as_bytes().to_vec();
+		if let Some(id) = IbcDenoms::<Runtime>::get(&denom_bytes) {
+			return Ok(id)
+		}
+
+		// let pallet_id: AccountId = PalletId(*b"pall-ibc").into_account_truncating();
```

Comment:
what is this?

Reply:
This code should be uncommented, it registers the native token as an asset on pallet assets

Reply:
@seunlanlege it doesn't work with the composable pallets, that's why I commented it and wrote new code below